### PR TITLE
refactor: create storage classes directly instead of using CRS

### PR DESCRIPTION
**Pull request by faiq**
**What problem does this PR solve?**:

**Which issue(s) this PR fixes**:
Removes our dependency for Cluster Resource Sets entirely for Nutanix Clusters

**How Has This Been Tested?**:

Create a cluster and check if storage class is created.

```sh
02:55 PM  faiqus @ archlinux  ~/go/src/github.com/d2iq-labs/cluster-api-runtime-extensions-nutanix  faiq/storageclass-direct? 
$ kubectl --kubeconfig=faiqcsi.conf get sc
NAME                       PROVISIONER       RECLAIMPOLICY   VOLUMEBINDINGMODE      ALLOWVOLUMEEXPANSION   AGE
nutanix-volume (default)   csi.nutanix.com   Delete          WaitForFirstConsumer   false                  3m

02:56 PM  faiqus @ archlinux  ~/go/src/github.com/d2iq-labs/cluster-api-runtime-extensions-nutanix  faiq/storageclass-direct? 
$ kubectl get clusterresourcesets
No resources found in default namespace.

02:56 PM  faiqus @ archlinux  ~/go/src/github.com/d2iq-labs/cluster-api-runtime-extensions-nutanix  faiq/storageclass-direct? 
$ kubectl get clusterresourcesets -A
No resources found

02:56 PM  faiqus @ archlinux  ~/go/src/github.com/d2iq-labs/cluster-api-runtime-extensions-nutanix  faiq/storageclass-direct? 
$ kubectl get helmreleaseproxy -A
NAMESPACE   NAME                                   CLUSTER   READY   REASON   STATUS     REVISION
default     cilium-faiqcsi-ndb7v                   faiqcsi   True             deployed   1
default     cluster-autoscaler-faiqcsi-v58lv       faiqcsi   True             deployed   1
default     node-feature-discovery-faiqcsi-q9n2w   faiqcsi   True             deployed   1
default     nutanix-cloud-provider-faiqcsi-2d4zq   faiqcsi   True             deployed   1
default     nutanix-csi-snapshot-faiqcsi-wjmjr     faiqcsi   True             deployed   1
default     nutanix-csi-storage-faiqcsi-6567j      faiqcsi   True             deployed   1
```

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/handlers/generic/lifecycle/utils/scs.go [hunk 1]
```diff
@@ -50,3 +57,38 @@ func CreateStorageClass(
...
+	}
+	clusterKey := ctrlclient.ObjectKeyFromObject(cluster)
+	remoteClient, err := remote.NewClusterClient(ctx, "", cl, clusterKey)
+	if err != nil {
+		return fmt.Errorf("error creating client for remote cluster: %w", err)
+	}
+	for _, o := range allStorageClasses {
+		err = client.ServerSideApply(ctx, remoteClient, o)
+		if err != nil {
+			return fmt.Errorf("error creating client for remote cluster: %w", err)
```
Comment (supershal) [inline pkg/handlers/generic/lifecycle/utils/scs.go:90]:
```suggestion
			return fmt.Errorf("error creating storage class %s for remote cluster:%s %w", <storage_class_name>,<cluster_name>, err)
```

**Review by supershal — APPROVED** [review]

File: pkg/handlers/generic/lifecycle/utils/scs.go [hunk 2]
```diff
@@ -50,3 +57,38 @@ func CreateStorageClass(
...
+
+func CreateStorageClassOnRemote(
+	ctx context.Context,
+	cl ctrlclient.Client,
+	configs []v1alpha1.StorageClassConfig,
+	cluster *clusterv1.Cluster,
+	defaultStorageConfig *v1alpha1.DefaultStorage,
+	csiProvider string,
+	provisioner v1alpha1.StorageProvisioner,
+	storageClassParameters map[string]string,
```
Comment (dkoshkin) [inline pkg/handlers/generic/lifecycle/utils/scs.go:69]: nit: matches what's in `CreateStorageClass`.
```suggestion
	defaultParameters map[string]string,
```

**Review by dkoshkin — APPROVED** [review]
Thanks for this change.

File: pkg/handlers/generic/lifecycle/utils/scs.go (hunk 2 above)
Comment (faiq) [reply pkg/handlers/generic/lifecycle/utils/scs.go]: done

**Review by dkoshkin — APPROVED** [review]

**Pull request merged**